streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -879,7 +879,10 @@ void to(final TopicNameExtractor<K, V> topicExtractor,
      * A {@link Transformer} (provided by the given {@link TransformerSupplier}) is applied to each input record and
      * returns zero or one output record.
      * Thus, an input record {@code <K,V>} can be transformed into an output record {@code <K':V'>}.
-     * This is a stateful record-by-record operation (cf. {@link #map(KeyValueMapper) map()}).
+     * This is a stateful record-by-record operation (cf. {@link #map(KeyValueMapper) map()}). Note that it is not
```

Comment:
I think this definitely helps clarify! A copy suggestion:
```
Attaching a state store makes this a stateful record-by-record operation (cf. {@link #map(KeyValueMapper) map()}).
If you choose not to attach a state store, this operation behaves similarly to {@link #map(KeyValueMapper) map()}
but additionally provides access to the {@code ProcessorContext} and associated metadata.
```

Reply:
Yeah, I wasn't quite sure how to phrase this because I wanted to keep the symmetry with the `#map` javadocs (eg `This is a stateless record-by-record...`) but this was actually the least-awkward wording I could come up with (and it's pretty awkward)
On second thought, that's kind of silly (and probably how we got into this mess in the first place) so I'll go with the more clear phrasing instead :)